### ISSUE TITLE
R0201 false positives

### ISSUE THREAD

**Issue by MatthijsBlom** [PyCQA/pylint#3281]
The following five valid special method names produce `R0201 Method could be a function` warnings.
```python
class C:
    def __pow__(self, other, modulo): return NotImplemented
    def __imatmul__(self, other): return NotImplemented
    def __trunc__(self): return NotImplemented
    def __floor__(self): return NotImplemented
    def __ceil__(self): return NotImplemented
```
Of these five, `__pow__` and `__imatmul__` also produce `W0613 Unused argument` warnings.

As far as I can tell, there are no other special method names that are not recognized as such.

I am currently looking into fixing this myself and may send a pull request in the near future.
